# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/nsIEditor.idl
# Commit: 71670cedaf26
# Full Hash: 71670cedaf26214f3bcb8562a445cf1ef24f4db6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-06-30 09:36:48
# Regressor Bug: 1714914
# File Overlap Count: 1
# Description:
#   Bug 1714914 - Make editor classes treat offset in parent node as uint32_t r=m_kato
#   
#   It should be treated as `uint32_t` since DOM API does so.  However, there are
#   some exceptions:
#   
# ==============================================================================

diff -r f5dcc2d793d6 -r 71670cedaf26 editor/nsIEditor.idl
--- a/editor/nsIEditor.idl	Tue Jun 29 22:00:09 2021 +0000
+++ b/editor/nsIEditor.idl	Wed Jun 30 07:07:28 2021 +0000
@@ -434,7 +434,7 @@
   [can_run_script]
   void insertNode(in Node node,
                   in Node parent,
-                  in long aPosition);
+                  in unsigned long aPosition);
 
 
   /**
@@ -543,9 +543,8 @@
 
   /**
     * The length of the contents in characters.
-    * XXX change this type to 'unsigned long'
     */
-  readonly attribute long textLength;
+  readonly attribute unsigned long textLength;
 
   /** Get and set the body wrap width.
     *